### PR TITLE
Added testing guidelines section to the CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by gasperzgonec**
## Summary

Added instructions for what tests should look like and how they should be implemented. This should help all developers working on this project to use consistent testing schema.

## Connected Issues

- [#ISS-202840](https://app.devrev.ai/devrev/works/ISS-202840)

### CODE REVIEW

**Review by patricijabrecko — COMMENTED** [review]
Do we also want to follow the AAA (Arrange, Act, Assert) pattern?

**Comment by gasperzgonec** [comment]
I've added AAA. Checking the current compliance, the AAA is being followed in about 85% of tests.
I'll update the remaining ones.

**Review by patricijabrecko — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -9,3 +9,120 @@ We follow Semantic Versioning (SemVer) for versioning, with the format `MAJOR.MI
...
+
+## Testing
+
+All new code must include comprehensive tests. Follow these testing guidelines:
+
+### Test File Structure
+
+- Test files must be named the same as the file containing the logic, but ending in `.test.ts`
+- Example: `user-service.ts` → `user-service.test.ts`
+- Place test files in the same directory as the source files
```
[the remark this answers is not in the document]
Comment (gasperzgonec) [reply CONTRIBUTING.md:21]: I didn't think of the integration tests, you're correct. 🤔

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -9,3 +9,120 @@ We follow Semantic Versioning (SemVer) for versioning, with the format `MAJOR.MI
...
+describe(ClassName.name, () => {
+  it('should perform expected behavior when given valid input', () => {
+    // Test implementation
+  });
+
+  it('should handle specific scenario correctly', () => {
+    // Test implementation
+  });
+
+  describe('[Edges]', () => {
```
[the remark this answers is not in the document]
Comment (gasperzgonec) [reply CONTRIBUTING.md:37]: Dejan said that separating edges from the other testing code is a nice way of splitting it up, as you have behaviour on one end, and the bugs/edges on the other.

[the remark this answers is not in the document]
Comment (Mesko89) [reply CONTRIBUTING.md:37]: Consider `[edge]` a tag instead. So that you can still use it in `it` as well as `describe`. I had an example with `describe('[edge]') because that is where they made more sense, but it is not wrong to use it in `it('[edge] should try to transform strings into numbers when string is provided', {})` for example.

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (gasperzgonec) [reply CONTRIBUTING.md:21]: What would you recommend to have there instead? @Mesko89

[the remark this answers is not in the document]
Comment (Mesko89) [reply CONTRIBUTING.md:21]: A nice example is metrics module on mobile-bff. We have file named `metrics.spec.ts` that tests the public facing (endpoint) controller while `metrics.interceptor.spec.ts` has tests specific for testing internal API that is used globally in our application.

Just be pragmatic, you can rename/move the test files later if the need arises or they organically do not fit in the same files anymore.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -9,3 +9,120 @@ We follow Semantic Versioning (SemVer) for versioning, with the format `MAJOR.MI
...
+      // Edge case test
+    });
+  });
+});
+```
+
+### Test Naming Conventions
+
+- **Describe blocks**: Use the class name being tested. You can use `<ClassName>.name` for consistency
+- **It blocks**: Write descriptive test names that start with "should" and describe the expected behavior
```
[the remark this answers is not in the document]
Comment (radovanjorgic) [reply CONTRIBUTING.md:50]: Sure, we added that too
